### ISSUE TITLE
[docs] Create a page about the Autosubmit architecture

### ISSUE THREAD

**Issue by kinow** [BSC-ES/autosubmit#1061]
We need a page to introduce the architecture of Autosubmit to users.

It must:

- explain the components of Autosubmit (database, job scheduler, experiment manager, etc.)
- user authentication (including how we authenticate to remote machines, which auth methods are used, etc.)
- demonstrate (in writing or images) why it's useful to use a separate host/VM for Autosubmit instead of deploying, for example, in the computing node

In summary, something users considering to use Autosubmit can refer to to get their basic questions answered.

**Comment by kinow** [comment]
Questions from another project about architecture of AS and API/GUI:

>For the API + network port, could you share the architecture, I'm a bit hesitant to open it wide could the GUI machine be guaranteed to have a static ip?. Also is it possible to run the API on a separate machine from where autosubmit itself is running?

I think the AS arch docs should either link to API/GUI arch docs, or have sub-sections for API and GUI.

**Comment by kinow** [comment]
And here's something useful I learned from an article I saved to read offline on a flight recently:

- [The C4 model for visualising software architecture](https://c4model.com/), Context, Containers, Components, and Code

The containers are not about containers as in Docker containers. It's a useful pattern to document and share info on architecture. Maybe we could use it for the AS docs. Will keep reading it over the next flights :grimacing: just documenting it here so I don't forget :wave: :beach:
